### ISSUE TITLE
Page creation validation doesn't recognize dashes 

### ISSUE THREAD

**Issue by perkinsjr** [tinacms/tinacms#2739]
## Description

Page creation supports 
` a-z, A-Z, 0-9, -, ., or _`

However currently dashes return form validation errors.

## Expected
Should be able to create with dashes

## Version
0.66.9

**Comment by julrich** [comment]
I think I just today stumbled upon this, too! I think this is what applies to field names, right?

We have some templates as part of an object type, those sometimes have names with dashes in them (like `visual-slider`). That currently breaks form generation for me! (right now hacking around it by just replacing those dashes, but would prefer not having to :) )

Error for us looks like this (only repeated for every occurrence of a dash in a name):
```
Names must match /^[_a-zA-Z][_a-zA-Z0-9]*$/ but "visual-slider" does not.
```

**Comment by Enigmatical** [comment]
Some good news on this front: 

We recently modified the Page Creation functionality to allow for sub-folders.  As part of that work, I redid the regex validation and, inadvertently, fixed this bug.

So apologies that you ran into this, but it should be fixed in 0.66.10 🎉
